build-script-helper.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+"""
+  This source file is part of the Swift.org open source project
+  Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+ ------------------------------------------------------------------------------
+ This is a helper script for the main swift repository's build-script.py that
+ knows how to build and install the stress tester utilities given a swift
+ workspace.
+"""
+
+from __future__ import print_function
+
+import argparse
+import sys
+import os, platform
+import subprocess
+
+def printerr(message):
+  print(message, file=sys.stderr)
+
+def main(argv_prefix = []):
+  args = parse_args(argv_prefix + sys.argv[1:])
+  run(args)
+
+def parse_args(args):
+  parser = argparse.ArgumentParser(prog='BUILD-SCRIPT-HELPER.PY')
+
+  parser.add_argument('--package-dir', default='')
+  parser.add_argument('-v', '--verbose', action='store_true', help='log executed commands')
+  parser.add_argument('--prefix', help='install path')
+  parser.add_argument('--config', default='debug')
+  parser.add_argument('--build-dir', default='.build')
+  parser.add_argument('--multiroot-data-file', help='Path to an Xcode workspace to create a unified build of SwiftSyntax with other projects.')
+  parser.add_argument('--toolchain', required=True, help='the toolchain to use when building this package')
+  parser.add_argument('--update', action='store_true', help='update all SwiftPM dependencies')
+  parser.add_argument('--no-local-deps', action='store_true', help='use normal remote dependencies when building')
+  parser.add_argument('build_actions', help="Extra actions to perform. Can be any number of the following", choices=['all', 'build', 'test', 'generate-xcodeproj'], nargs="*", default=['build'])
+
+  parsed = parser.parse_args(args)
+
+  parsed.swift_exec = os.path.join(parsed.toolchain, 'bin', 'swift')
+
+  # Convert package_dir to absolute path, relative to root of repo.
+  repo_path = os.path.dirname(__file__)
+  parsed.package_dir = os.path.realpath(
+                        os.path.join(repo_path, parsed.package_dir))
+
+  # Convert build_dir to absolute path, relative to package_dir.
+  parsed.build_dir = os.path.join(parsed.package_dir, parsed.build_dir)
```

Comment:
This is unintuitive to me.  In CI, this should always be absolute, and it should never be inside your source directory.  If I run this script manually, I would expect it to be relative to my working directory not the source tree.  If you want to make the default value be $SRC/.build to match swiftpm, you should only apply that to the default, not to all paths.  I.e.

* change default to `None`
* If path is `None`, explicitly set `build_dir = os.path.join(parsed.package_dir, '.build')`